src/app/plugins/cat.plugin.ts

```diff
@@ -0,0 +1,105 @@
+import Constants from '../../common/constants';
+import { Plugin } from '../../common/plugin';
+import { ChannelType, IContainer, IHttpResponse, IMessage } from '../../common/types';
+
+class Breed {
+    public breed: string = '';
+    public id: string = '';
+  }
+
+export class CatPlugin extends Plugin {
+  public name: string = 'Cat Plugin';
+  public description: string = 'Generates pictures of cats, via popular command.';
+  public usage: string = 'cat breed or pic bomb(optional) <breed (optional)>';
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName: string = Constants.Channels.Public.Pets;
+
+  private _API_URL: string = 'https://api.thecatapi.com/v1/';
+  private _breadCat: string = `https://www.thetoychronicle.com/wp-content/uploads/2016/05/Box-cat-Bread-cat-Rato-workroom-Kim-.jpg`;
+  private _cats: Breed[] = [];
+
+  constructor(public container: IContainer) {
+    super();
+    this.container.httpService
+      .get(`${this._API_URL}breeds`)
+      .then((response: IHttpResponse) => {
+        const breeds = response.data;
+
+        for (let index = 0; index < breeds.length; index++) {
+
+            const breedInfo = {
+                breed: breeds[index].name.toLowerCase(),
+                id: breeds[index].id.toLowerCase(),
+            };
+
+            this._cats.push(breedInfo);
+        }
+      })
+      .catch((err) => console.log(err));
+  }
+
+  public async execute(message: IMessage, args?: string[]) {
+    
+    let command = this._parseCommand(args || []);
+    
+    if (command === "breeds" || command === "breed") {
```

Comment:
before transforming `args` into a string, we can do a sanity check to check if `args[0] === 'breeds'`, if it returns false, then we can build the breed string